packages/react/src/button-group/ButtonGroup.tsx

```diff
@@ -0,0 +1,48 @@
+import clsx from "clsx";
+import {
+  Children,
+  type ComponentPropsWithRef,
+  type ReactNode,
+  forwardRef,
+} from "react";
+
+import type { ExtendProps } from "../utils";
+
+import { Box } from "../box";
+import { Flex } from "../flex";
+import * as styles from "./ButtonGroup.css";
+
+type ButtonGroupProps = ExtendProps<
+  ComponentPropsWithRef<typeof Flex>,
+  {
+    children: ReactNode;
+  } & styles.ButtonGroupVariants
+>;
+
+export const ButtonGroup = forwardRef<HTMLDivElement, ButtonGroupProps>(
+  (
+    { children, className, gap = "0", orientation = "horizontal", ...props },
+    ref,
+  ) => {
+    const mappedChildren = Children.map(children, (child) => (
+      <Box
+        asChild
+        className={styles.button({ orientation, spacing: gap !== "0" })}
```

Comment:
Let's forward some specific props to each child button:

- `appearance`
- `colorScheme`
- `disabled`
- `size`
- `variant`

So the idea is you can set those properties for all children buttons by setting them at the group level:

```
<ButtonGroup size="sm">
</ButtonGroup>
```

This will set the (same) size for all child buttons.

Reply:
instead of `appearance` it should be `preset` bhai

Reply:
addressed bhai